### PR TITLE
[Native] Fix PlanNodeSerdeTest unit tests

### PR DESCRIPTION
**Pull request by gggrace14**
Two unit tests
partitionAndSerializeNode
shuffleWriteNode
got accidentally broken by presto PR https://github.com/prestodb/presto/pull/19554 picking up Velox PR https://github.com/facebookincubator/velox/pull/4670.

This is a followup fix. Please rebase and include this fix if PlanNodeSerdeTest unit tests are broken.

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-native-execution/presto_cpp/main/operators/tests/PlanNodeSerdeTest.cpp [hunk 1]
```diff
@@ -104,7 +102,9 @@ class PlanNodeSerdeTest : public testing::Test,
     Type::registerSerDe();
     core::PlanNode::registerSerDe();
     core::ITypedExpr::registerSerDe();
-    exec::registerPartitionFunctionSerDe();
+    auto& registry = DeserializationWithContextRegistryForSharedPtr();
```
Comment (mbasmanova) [inline presto-native-execution/presto_cpp/main/operators/tests/PlanNodeSerdeTest.cpp:105]: This doesn't look right. 

CC: @kewang1024 @xiaoxmeng

**Comment by mbasmanova** [comment]
@gggrace14 Ge, what is the failure you are fixing? Is there a GitHub issue? How did this slip through CI?

File: presto-native-execution/presto_cpp/main/operators/tests/PlanNodeSerdeTest.cpp (hunk 1 above)
Comment (gggrace14) [reply presto-native-execution/presto_cpp/main/operators/tests/PlanNodeSerdeTest.cpp:105]: Hi Masha @mbasmanova , with exec::registerPartitionFunctionSerDe(), PlanNodeSerdeTest in presto_cpp has two unit test failed with error

"couldn't find key keyChannels in dynamic object"

I got the above piece of logic from Ke, but forgot to include it in the "git commit --amend" in my Velox version advance PR  https://github.com/prestodb/presto/pull/19554. So adding it in this separate PR. This logic fixes the two unit test error on my local.

Comment (mbasmanova) [reply presto-native-execution/presto_cpp/main/operators/tests/PlanNodeSerdeTest.cpp:105]: @kewang1024 @xiaoxmeng  Ke, why exec::registerPartitionFunctionSerDe() is not working?

@gggrace14 Ge, what's the error? We should not "manually" register these classes, rather use helper methods like exec::registerPartitionFunctionSerDe().

Comment (gggrace14) [reply presto-native-execution/presto_cpp/main/operators/tests/PlanNodeSerdeTest.cpp:105]: Masha, the error is just

*"couldn't find key keyChannels in dynamic object"*

when we run the partitionAndSerializeNode and shuffleWriteNode tests. It's reproducible locally.

Comment (kewang1024) [reply presto-native-execution/presto_cpp/main/operators/tests/PlanNodeSerdeTest.cpp:105]: So yesterday we merged the refactoring on velox, but haven't started the change on presto native but want to unblock the build

So we have several same spec definition: 
1. In presto native test (Needs to be removed)
2. In velox test (Just removed)
3. In velox prod (Newly introduced)

So this is the 1, and this change would be temporary

Comment (mbasmanova) [reply presto-native-execution/presto_cpp/main/operators/tests/PlanNodeSerdeTest.cpp:105]: I reproduced the failure locally. I believe the right fix is to delete HashPartitionFunctionSpec class from this test file (and use the one defined in velox/exec/HashPartitionFunction.h).

File: presto-native-execution/presto_cpp/main/operators/tests/PlanNodeSerdeTest.cpp [hunk 2]
```diff
@@ -104,7 +40,9 @@ class PlanNodeSerdeTest : public testing::Test,
...
```
Comment (kewang1024) [inline presto-native-execution/presto_cpp/main/operators/tests/PlanNodeSerdeTest.cpp:43]: #include "velox/exec/PartitionFunction.h"

exec::registerPartitionFunctionSerDe();

**Review by kewang1024 — COMMENTED** [review]
Thanks Ge for helping make this change!

File: presto-native-execution/presto_cpp/main/operators/tests/PlanNodeSerdeTest.cpp (hunk 2 above)
Comment (gggrace14) [reply presto-native-execution/presto_cpp/main/operators/tests/PlanNodeSerdeTest.cpp]: Thank you for reminding, Ke! Doing the change all over again.

**Comment by gggrace14** [comment]
Test failures as below for this version are for coordinator and not related. Native e2e tests have not been run, since they are after the first presto-druid dep failure below. The native unit tests passed. As this PR is a fix to unit test, no run of e2e should be fine?

1. Could not resolve dependencies for project com.facebook.presto:presto-druid

2. Error:  Failures: 
Error:    TestAtopSmoke.testReboots:62->assertThatQueryReturnsValue:72 expected [2] but found [0]
[INFO] 
Error:  Tests run: 10, Failures: 1, Errors: 0, Skipped: 0

**Review by mbasmanova — COMMENTED** [review]
@gggrace14 Ge, the fix looks good to me, but Ci is red. Let's dig in into what's still broken. Have you already tried rebasing onto latest master?

CC: @arunthirupathi @feilong-liu

**Comment by arunthirupathi** [comment]
this does not contain the fix for broken master, https://github.com/gggrace14/presto/commits/ebadb39193d046105ce5a505a7d09c806ae9d4d8

Since this is a native test only change, as long as native test passes, we are good to go on this one. I have restarted the failed native tests.

**Comment by gggrace14** [comment]
> @gggrace14 Ge, the fix looks good to me, but Ci is red. Let's dig in into what's still broken. Have you already tried rebasing onto latest master?
> 
> CC: @arunthirupathi @feilong-liu

Hi Masha, yes, I rebased onto the latest master by last night

**Comment by gggrace14** [comment]
> this does not contain the fix for broken master, https://github.com/gggrace14/presto/commits/ebadb39193d046105ce5a505a7d09c806ae9d4d8
> 
> Since this is a native test only change, as long as native test passes, we are good to go on this one. I have restarted the failed native tests.

Thank you for info, @arunthirupathi :)

**Review by kewang1024 — APPROVED** [review]
Thanks Ge for help making this change!

**Review by mbasmanova — APPROVED** [review]

**Pull request merged**